first_problem_test.rb

```diff
@@ -0,0 +1,39 @@
+require 'test/unit'
+require_relative'./first_problem'
+
+class FirstProblemTest < Test::Unit::TestCase
+
+  # Called before every test method runs. Can be used
+  # to set up fixture information.
+  def setup
+    # Do nothing
+  end
+
+  # Called after every test method runs. Can be used to tear
+  # down fixture information.
+
+  def teardown
+    # Do nothing
+  end
+
+  def test_compute_emptyArray_0
+    array = Array.new
+    assert_equal(0,FirstProblem.new(array).compute)
+  end
+
+  def test_compute_noEvenNumbersArray_0
+    array = [1]
+    assert_equal(0,FirstProblem.new(array).compute)
```

Comment:
You should have a separate unit test for each of the asserts here

Reply:
Why ? The asserts are against the expected input. 
Both the arrays have no even numbers

Reply:
There a couple of reasons why having multiple asserts in a tests is not a good idea. The most obvious one is that you where asked to do tdd, that means write a test that fails (and it will fail after the first assert), fix it (the minimum code required to make the test pass) and then refactor.

Reply:
I guess i didn't quite understand that you wanted me to do 1 failing unit test , 1 code change , 1 refactor (if neccesary) at a time.
I did more of a add unit tests (failing) for functionality , do a "demo" version for 1st one passing then 2/3rd failing . 
If i was developing TDD i would imagine i don't commit every time i have created 1 failing unit test. To then commit again with it working. To then commit again when refactor. At least in my mind i would commit when it's done once.
But this is just how i envisioned real working scenario. I could be totally wrong. And that's why all my PR's look like that.

Related to having one assert per unit test , in this particular case , i could have just merged them. There is no correlation between size of array and output.
I agree that maybe it could be confusing to have 2 assert for the same validation.